doc/source/user/basics.broadcasting.rst

```diff
@@ -184,4 +177,75 @@ Here the ``newaxis`` index operator inserts a new axis into ``a``,
 making it a two-dimensional ``4x1`` array.  Combining the ``4x1`` array
 with ``b``, which has shape ``(3,)``, yields a ``4x3`` array.
 
-
+A Practical Example: Vector Quantization
+========================================
+
+Broadcasting comes up quite often in real world problems. A typical example
+occurs in the vector quantization (VQ) algorithm used in information theory,
+classification, and other related areas. The basic operation in VQ finds
+the closest point in a set of points, called ``codes`` in VQ jargon, to a given
+point, called the ``observation``. In the very simple, two-dimensional case
+shown below, the values in ``observation`` describe the weight and height of an
+athlete to be classified. The ``codes`` represent different classes of
+athletes. [#f1]_ Finding the closest point requires calculating the distance
+between observation and each of the codes. The shortest distance provides the
+best match. In this example, ``codes[0]`` is the closest class indicating that
+the athlete is likely a basketball player.
+
+  >>> from numpy import array, argmin, sqrt, sum
+  >>> observation = array([111.0, 188.0])
+  >>> codes = array([[102.0, 203.0],
+  ...                [132.0, 193.0],
+  ...                [45.0, 155.0],
+  ...                [57.0, 173.0]])
+  >>> diff = codes - observation    # the broadcast happens here
+  >>> dist = sqrt(sum(diff**2,axis=-1))
+  >>> argmin(dist)
+  0
+
+In this example, the ``observation`` array is stretched to match
+the shape of the ``codes`` array::
+
+  Observation      (1d array):      2
+  Codes            (2d array):  4 x 2
+  Diff             (2d array):  4 x 2
+
+.. figure:: theory.broadcast_5.png
+    :alt: vector quantitization example
+    :name: figure-1
```

Comment:
This is the reason for the CI failure: this image name is referenced twice (here and in the other document). Can you change this "name" argument to something different?

Reply:
Thanks! I was aware of that but decided to ignore it for now because eventually the other doc would be removed and it wouldn't clash anymore.  But now that I think of it, I should use a different naming system e.g. "doc-name.figure no." so that it doesn't cause a problem later on when we include figures in other documents too.